core/communication_manager.py

```diff
@@ -80,17 +81,15 @@ def inform(self, event_type, payload):
         if self.optimizer:
             # We have an active session.
             event_type, payload = self.optimizer.ask(event_type, payload)
-            callback = callback_handler(
+            callback = callback_handler_no_default(
```

Comment:
Why did you make this change? Did you mean to do it in L90?

The current design says that the optimizer could inform the Communication Manager to do nothing if that's its policy.

Reply:
true, neither this nor L90 should be w/o default

Reply:
I'm also OK with removing all NOTHINGs everywhere (and `callback_handler()` calls) and not worrying about it for now since we're only supporting one Optimizer at the moment anyways. This would probably make things less error-prone and simpler.

Reply:
Cool, seems we're on the same page with this then.

Reply:
To be clear, what I'm saying is either:

1- L84 using `callback_handler()` and L90 using `callback_handler_no_default()`; or,
2- L84 and L90 using `callback_handler_no_default()`

(If you decide otherwise, then this is dependent on the other thread we're discussing in `enums.py`'s L54 and we should address this thread after that one is solved.)